client/my-sites/design-preview/index.js

```diff
@@ -165,36 +164,52 @@ const DesignPreview = React.createClass( {
 			return;
 		}
 		event.preventDefault();
-		// TODO: if the href is on the current site, load the href as a preview and fetch markup for that url
 	},
 
-	getPreviewUrl( site ) {
+	getParsedPreviewUrl( site ) {
 		if ( ! site ) {
 			return null;
 		}
+		const baseUrl = this.props.previewUrl ? this.props.previewUrl : site.options.unmapped_url;
+		const parsed = url.parse( baseUrl, true );
+		delete parsed.search;
+		return parsed;
+	},
 
-		const parsed = url.parse( site.options.unmapped_url, true );
+	getPreviewUrl( site ) {
+		if ( ! site ) {
```

Comment:
Similar to above, should we move this condition a few lines down to simply guard against adding the `frame-nonce`? Or am I misunderstanding the requirements on having a site here?